### ISSUE TITLE
Parsing of a table returns a garbage property.

### ISSUE THREAD

**Issue by nareshbhatia** [acuminous/yadda#180]
Given the following step:
```
When I create a transaction with the following properties
    -------------------------------------------------------------------------------------
    | date       | payee               | memo | amount | account     | category         |
    | 2015-02-01 | Chevron Gas Station | Gas  | -30.00 | Amazon VISA | Auto & Transport |
    -------------------------------------------------------------------------------------
```

And the following step definition:

```
.when('I create a transaction with the following properties\n$table', function(table, next) {
    console.log(table[0]);
    next();
})
```

And the following table converter
```
var dictionary = new Dictionary()
    .define('table', /([^\u0000]*)/, Yadda.converters.table);
```

I get the following console log:

```
{ '': '\n',
  date: '2015-02-01',
  payee: 'Chevron Gas Station',
  memo: 'Gas',
  amount: '-30.00',
  account: 'Amazon VISA',
  category: 'Auto & Transport'
}
```

1. Note that there is an extra property in the output with a value of '\n'. How can this property be eliminated when parsing tables?
2. Can you please explain how the regular expression `/([^\u0000]*)/` works to pick up the table and only the table. I am reading it as pickup all non-null characters. Does that mean span multiple lines? If yes, then it would pick up the rest of the document, which is not what we want. If not, it will pick up only one row of the table, which is also not what I want!

**Comment by nareshbhatia** [comment]
I am seeing the same issues with lists - an extra blank property:

```
[ '', 'name', 'Cash', 'Credit' ]
```

**Comment by nareshbhatia** [comment]
Of course, when I try your data-tables example, it works perfectly. List shows up as:

```
[ '1000', '2022', '9405' ]
```

Table shows up as:

```
[ { left: '1', right: '3' }, { left: '2', right: '4' } ]
```

No extra elements! Puzzled what could be different in my example. Here's my [source](https://github.com/archfirst/acceptance-testing-best-practices/tree/master/mocha-yadda-supertest/test).

Naresh

**Comment by nareshbhatia** [comment]
@cressie176, any update on this?

**Comment by cressie176** [comment]
Not had a chance to look at it yet. Sorry.

**Comment by cressie176** [comment]
Found the problem.

https://github.com/archfirst/acceptance-testing-best-practices/blob/master/mocha-yadda-supertest/test/steps/accounts.steps.js#L70

```
'the following accounts\$list'
```
Should be
```
'the following accounts\n$list'
```

**Comment by nareshbhatia** [comment]
Embarrassed :-$. My bad for not seeing this.

This leaves the issue with table parsing. The details are in the very first comment of this issue, but here's a summary:

1. When parsing a table (see example [here](https://github.com/archfirst/acceptance-testing-best-practices/blob/master/mocha-yadda-supertest/test/steps/transactions.steps.js#L47-L55)), I am getting an extra property for each row: `'': '\n'`. How can this property be eliminated? Again, your data-tables example is not showing this behavior.
2. Can you please explain how the regular expression `/([^\u0000]*)/` works to pick up the table and only the table. I am reading it as pickup all non-null characters. Does this expression span multiple lines? If yes, then it should pick up the rest of the document, which is not what we want. If not, it should pick up only the first line of the table, which is also not what we want!

**Comment by cressie176** [comment]
> When parsing a table (see example here), I am getting an extra property for each row: '': '\n'. How can this property be eliminated? Again, your data-tables example is not showing this behavior.

This is fixed by adding the \n to your step text

> Can you please explain how the regular expression /([^\u0000]*)/ works to pick up the table and only the table. I am reading it as pickup all non-null characters. Does this expression span multiple lines? If yes, then it should pick up the rest of the document, which is not what we want. If not, it should pick up only the first line of the table, which is also not what we want!

The Feature parser reads the feature file line by line. When the parser detects a multiline table it appends all rows into a single step. The regular expression in your step text therefore needs to be able to span multiple lines, but it since it only operates on individual steps rather than the whole document it can't match text beyond the table.

**Comment by nareshbhatia** [comment]
> This is fixed by adding the \n to your step text

That issue was only with the list converter, because I had forgotten the \n in my regular expression. The table step always had a \n. See [here](https://github.com/archfirst/acceptance-testing-best-practices/blob/master/mocha-yadda-supertest/test/steps/transactions.steps.js#L47) for an example. I am still seeing the issue in spite of that. So I think these are unrelated problems. Please let me know if I am misunderstanding your comment.

**Comment by cressie176** [comment]
The data table parse was treating outer borders as new columns. Fixed in 0.16.0

**Comment by nareshbhatia** [comment]
Thanks, @cressie176. This was the last of the remaining issues for me to "officially" switch over to Yadda, which I have done now. It's been a great learning experience and I am really impressed by what it can do. Thanks for all your efforts. Please feel free to include my [example](https://github.com/archfirst/acceptance-testing-best-practices/tree/master/mocha-yadda-supertest) in your documentation. I think it will be very useful for people like me who want to use Yadda for testing complex applications. Key features of this example are:
- It tests a realistic application called *Manage My Money* (a person finance manager).
- It uses multiple feature files and step files.
- It uses lists and data tables.
- It integrates with [SuperTest](https://github.com/visionmedia/supertest) to drive a REST API.
- It uses Mocha before and after hooks to initialize and release database connections.
